browse/controllers/list_page/__init__.py

```diff
@@ -0,0 +1,424 @@
+"""Handle requests for the /list pages.
+
+/list requests will show a list of articles for a category for a given
+time period.
+
+The primary entrypoint to this module is :func:`.get_list_page`, which
+handles GET and POST requests to the list endpoint.
+
+This should handle requests like:
+/list/$category/YYMM
+/list/$category/YYMM
+/list/category/recent
+/list/category/YYMM?skip=n&show=n
+/list/archive/new|recent|pastweek
+/list/archive/YY
+/list/$category/YY
+
+Examples of odd requests to throw out:
+/list/?400
+/list/cs/14?skip=%25CRAZYSTUFF
+/list/1801.00023
+
+1. Figure out what category and time_period is being requested. It's
+either a POST or GET with params about what to get OR it's all in the
+path.
+
+Things to figure out:
+A: what subject category is being requested
+B: time period aka listing_type: 'pastweek' 'new' 'current' 'pastyear'
+C: show_abstracts only if listing_type='new'
+
+2. Query the listing service for that category and time_period
+
+3. Check for not modified.
+
+4. Disply the page
+
+Differences from legacy arxiv:
+Doesn't handle the /view path.
+"""
+import calendar
+from typing import Any, Dict, List, Optional, Tuple, cast
+import math
+
+from flask import current_app, request, url_for
+
+from werkzeug.exceptions import ServiceUnavailable, BadRequest
+from arxiv import status, taxonomy
+
+from browse.services.search.search_authors import queries_for_authors, \
+    split_long_author_list, AuthorList
+from browse.services.listing import ListingService, NewResponse
+from browse.services.document import metadata
+from browse.domain.metadata import DocMetadata
+from browse.controllers.abs_page import truncate_author_list_size
+from browse.controllers.list_page.paging import paging
+
+
+show_values = [5, 10, 25, 50, 100, 250, 500, 1000, 2000]
+"""" Values of $show for more/fewer/all."""
+
+max_show = show_values[-1]
+"""Max value for show that controller respects."""
+
+default_show = show_values[2]
+"""Default value for show."""
+
+Response = Tuple[Dict[str, Any], int, Dict[str, Any]]
+
+type_to_template = {
+    'new': 'list/new.html',
+    'recent': 'list/recent.html',
+    'current': 'list/month.html',
+    'month': 'list/month.html',
+    'year': 'list/year.html'
+}
+
+
+def get_listing(
+        subject_or_category: str,
+        time_period: str,
+        skip: str,
+        show: str) -> Response:
```

Comment:
Curious why skip and show are of type `str`?

Reply:
They are str because that's what seemed to be getting passed from Flask.